lib/ansible/modules/cloud/amazon/ec2_eip.py

```diff
@@ -411,8 +410,12 @@ def main():
                 result = ensure_present(ec2, module, domain, address, private_ip_address, device_id,
                                     reuse_existing_ip_allowed, module.check_mode, isinstance=is_instance)
             else:
-                address = allocate_address(ec2, domain, reuse_existing_ip_allowed)
-                result = {'changed': True, 'public_ip': address.public_ip, 'allocation_id': address.allocation_id}
+                address = find_address(ec2, public_ip, None)
```

Comment:
Since this is now happening here and on [line 426](https://github.com/ansible/ansible/pull/30322/files#diff-cc1ca32d3b556628c6b2562406c7478dL426) does it make sense to change [line 407](https://github.com/ansible/ansible/pull/30322/files#diff-cc1ca32d3b556628c6b2562406c7478dL407) instead? Smaller net diff. Besides that, tested this and works for me.

Reply:
Yes, that does make more sense. Good spot, thanks.